### PR TITLE
Feat/translations

### PR DESCRIPTION
**Pull request by durbar**
Description:
Describe your pull request and attach links to any issues that may be related.

### CODE REVIEW

File: client/js/main.js [hunk 1]
```diff
@@ -3876,6 +3877,772 @@ class FormatSQL {
...
+
+		const container = document.createElement('div');
+		container.classList.add('add-translations-container');
+
+		container.id = 'phrases-translations';
+
+		container.innerHTML = `
+			<div id="list-container">
+			
+				<button class="Add"> Add<i class="fa fa-paper-plane"></i></button>
```
Comment (achshar) [inline client/js/main.js]: Icon before text.

File: client/js/main.js [hunk 2]
```diff
@@ -3876,6 +3877,772 @@ class FormatSQL {
...
+				<form class="form block" id="add-translation-form">
+					<label>
+						<span> Phrase</span>
+						<input type="text" name="phrase" required>
+					</label>
+					<label class="locale-select">
+						<span> Locale</span>
+					</label>
+					<label>
+						<span> Translation</span>
```
Comment (achshar) [inline client/js/main.js]: Space after opening span not needed

File: client/js/main.js [hunk 3]
```diff
@@ -3876,6 +3877,772 @@ class FormatSQL {
...
+					</div>
+					<div id="translations-advanced-search"></div>
+					<div id="translations-headers"></div>
+				</div>
+			</div>
+			
+			<div class="hidden" id="add-translation-form-container">
+				<div class="toolbar translations-list-view">
+					<label>
+						<input type="button" name="back" value="back">
```
Comment (achshar) [inline client/js/main.js]: Use `<button>` and add icon

File: client/js/main.js [hunk 4]
```diff
@@ -3876,6 +3877,772 @@ class FormatSQL {
...
+		if(this.owner != 'phrase') {
+
+			fields.splice(0, 2);
+			const node = container.querySelector('input[name=phrase]').parentElement;
+			node.parentNode.removeChild(node);
+		}
+
+		for(const header of fields) {
+
+			const div = document.createElement('div');
```
Comment (achshar) [inline client/js/main.js]: use `insertAdjacentHTML`

File: client/js/main.js [hunk 5]
```diff
@@ -3876,6 +3877,772 @@ class FormatSQL {
...
+			div.textContent = header;
+			div.classList.add('translation-headers');
+			translationList.appendChild(div);
+		}
+
+		this.translationRowGridTemplateColumns = new Array(fields.length - 2).fill('1fr').concat('100px', '100px').join(' ');
```
Comment (achshar) [inline client/js/main.js]: use classes to define two different css rules.

File: client/js/main.js [hunk 6]
```diff
@@ -3876,6 +3877,772 @@ class FormatSQL {
...
+
+		translationList.style.gridTemplateColumns = this.translationRowGridTemplateColumns;
+
+		container.id = 'phrases-translations';
+
+		const selectLabel = container.querySelector('.locale-select');
+
+		selectLabel.appendChild(this.select());
+
+		container.querySelector('.Add').addEventListener('click', e => {
```
Comment (achshar) [inline client/js/main.js]: lowercase

File: client/js/main.js [hunk 7]
```diff
@@ -3876,6 +3877,772 @@ class FormatSQL {
...
+
+	get expandedContainer() {
+
+		const container = document.createElement('div');
+		container.classList.add('add-translations-container');
+
+		container.innerHTML = `
+			<div id="list-container">
+
+				<button class="Add"> Add<i class="fa fa-paper-plane"></i></button>
```
Comment (achshar) [inline client/js/main.js]: Icon before text

File: client/js/main.js [hunk 8]
```diff
@@ -3876,6 +3877,772 @@ class FormatSQL {
...
+
+		for(const header of fields) {
+
+			const div = document.createElement('div');
+			div.textContent = header;
+			div.classList.add('translation-headers');
+			translationList.appendChild(div);
+		}
+
+		this.translationRowGridTemplateColumns = new Array(fields.length - 2).fill('1fr').concat('100px', '100px').join(' ');
```
Comment (achshar) [inline client/js/main.js]: use css

File: client/js/main.js [hunk 9]
```diff
@@ -3876,6 +3877,772 @@ class FormatSQL {
...
+		translationList.style.display = 'grid';
+		translationList.style['grid-template-columns'] = this.translationRowGridTemplateColumns;
+
+		container.id = 'phrases-translations';
+
+		const selectLabel = container.querySelector('.locale-select');
+
+		selectLabel.appendChild(this.select());
+
+		container.querySelector('.Add').addEventListener('click', async e => {
```
Comment (achshar) [inline client/js/main.js]: lowercase

File: client/js/main.js [hunk 10]
```diff
@@ -3876,6 +3877,772 @@ class FormatSQL {
...
+
+	async load() {
+
+		await this.fetch();
+		await this.render();
+	}
+
+	async fetch() {
+
+		const params = {};
```
Comment (achshar) [inline client/js/main.js]: parameters

File: client/js/main.js [hunk 11]
```diff
@@ -3876,6 +3877,772 @@ class FormatSQL {
...
+		params.owner = this.owner || 'phrase';
+
+		if(this.owner_id) {
+
+			params.owner_id = this.owner_id;
+		}
+
+		this.response = await API.call('translations/list', params);
+
+		this.response = this.response.sort((x, y) => x.id - y.id);
```
Comment (achshar) [inline client/js/main.js]: not needed

File: client/js/main.js [hunk 12]
```diff
@@ -3876,6 +3877,772 @@ class FormatSQL {
...
+			e.preventDefault();
+
+			container.querySelector('.phrase-editor').parentNode.classList.add('hidden');
+
+			container.querySelector('#list-container').classList.add('hidden');
+			container.querySelector('#add-translation-form-container').classList.remove('hidden');
+
+			const node = container.querySelector('.translation-editor');
+
+			node.parentNode.style.maxWidth = 'unset';
```
Comment (achshar) [inline client/js/main.js]: use css

File: client/js/main.js [hunk 13]
```diff
@@ -3876,6 +3877,772 @@ class FormatSQL {
...
+
+		container.id = 'phrases-translations';
+
+		const selectLabel = container.querySelector('.locale-select');
+
+		selectLabel.appendChild(this.select());
+
+		container.querySelector('.Add').addEventListener('click', async e => {
+
+			e.preventDefault();
```
Comment (achshar) [inline client/js/main.js]: Not needed

File: client/js/main.js [hunk 14]
```diff
@@ -3876,6 +3877,772 @@ class FormatSQL {
...
+
+	async render() {
+
+		const listContainer = this.container.querySelector('#phrases-translations-existing');
+
+		const rows = listContainer.querySelectorAll('.translation-row');
+
+		for(const element of rows) {
+
+			if(!element.id) {
```
Comment (achshar) [inline client/js/main.js]: Not needed

File: client/js/main.js [hunk 15]
```diff
@@ -3876,6 +3877,772 @@ class FormatSQL {
...
+
+		for (const locale of MetaData.locales.values()) {
+
+			const option = document.createElement('option');
+			option.value = locale.id;
+			option.text = `${locale.name} (${locale.locale})`;
+
+			selectContainer.appendChild(option);
+
+			if (locale.id == selected) {
```
[the remark this answers is not in the document]
Comment (durbar) [reply client/js/main.js]: option.selected then be changed to querySelector, querySelector in a loop.

File: client/js/settings.js [hunk 16]
```diff
@@ -3142,6 +3342,199 @@ class CachedReport {
...
+	get row() {
+
+		if (this.getRow) {
+
+			return this.getRow;
+		}
+
+		const container = document.createElement('tr');
+
+		const locale = MetaData.locales.get(this.locale_id);
```
Comment (achshar) [inline client/js/settings.js]: Move to this.

File: client/js/settings.js [hunk 17]
```diff
@@ -3142,6 +3342,199 @@ class CachedReport {
...
+
+	constructor(row, page) {
+
+		Object.assign(this, row);
+		this.page = page;
+	}
+
+	get row() {
+
+		if (this.getRow) {
```
Comment (achshar) [inline client/js/settings.js]: `rowElement`

File: client/js/settings.js [hunk 18]
```diff
@@ -3142,6 +3342,199 @@ class CachedReport {
...
+		}
+
+		const container = document.createElement('tr');
+
+		const locale = MetaData.locales.get(this.locale_id);
+
+		container.insertAdjacentHTML('beforeend', `
+			<td>${this.id}</td>
+			<td>${this.phrase}</td>
+			<td>${locale.name}(${locale.locale})</td>
```
Comment (achshar) [inline client/js/settings.js]: Space after name

File: client/js/settings.js [hunk 19]
```diff
@@ -3142,6 +3342,199 @@ class CachedReport {
...
+			<td class="action red" title="Delete"><i class="far fa-trash-alt"></i></td>
+		`);
+
+		container.querySelector('.green').addEventListener('click', async () => {
+
+			await Sections.show(`edit-translation-form-${this.id}`);
+		});
+
+		container.querySelector('.red').removeEventListener('click', RowTranslations.deleteEventListener);
+		container.querySelector('.red').addEventListener('click', RowTranslations.deleteEventListener = async () => {
```
Comment (achshar) [inline client/js/settings.js]: `deleteEventListener` Not needed

File: db-schema/allspark.sql [hunk 20]
```diff
@@ -486,6 +486,46 @@ CREATE TABLE `tb_visualizations` (
   PRIMARY KEY (`id`)
 ) ENGINE=InnoDB DEFAULT CHARSET=latin1;
 
+
+CREATE TABLE `tb_documentation` (
```
Comment (achshar) [inline db-schema/allspark.sql]: Remove Documentation

**Pull request merged**